pkg/ecdsa/retry/retry.go

```diff
@@ -83,3 +83,210 @@ func EvaluateRetryParticipantsForSigning(
 	}
 	return seats, nil
 }
+
+func excludeSingleOperator(
+	rng *rand.Rand,
+	groupMembers []chain.Address,
+	index int,
+	operatorToSeatCount map[chain.Address]uint,
+	operators []chain.Address,
+) ([]chain.Address, int, bool) {
+	if index < len(operators) {
+		rng.Shuffle(len(operators), func(i, j int) {
+			operators[i], operators[j] = operators[j], operators[i]
+		})
+		removedOperator := operators[index]
+		usedOperators := make([]chain.Address, 0, len(groupMembers))
+		for _, operator := range groupMembers {
+			if operator != removedOperator {
+				usedOperators = append(usedOperators, operator)
+			}
+		}
+		return usedOperators, 0, true
+	} else {
+		return nil, len(operators), false
+	}
+}
+
+func excludeOperatorPairs(
+	rng *rand.Rand,
+	groupMembers []chain.Address,
+	index int,
+	operatorToSeatCount map[chain.Address]uint,
+	operators []chain.Address,
+	retryParticipantsCount int,
+) ([]chain.Address, int, bool) {
+	pairIndexes := make([][2]int, 0, len(operators)*len(operators))
```

Comment:
This slice contains only unique pairs right?

Reply:
Yeah - because of the way the for loop is structured, they'll be unique including ordering!

For example, here's some output:

```go
nums := [5]int{0,1,2,3,4}
for i := 0; i < len(nums) - 1; i++ {
	for j := i+1; j < len(nums); j++ {
		fmt.Printf("%d, %d\n", i, j)
	}
}
	
produces
0, 1
0, 2
0, 3
0, 4
1, 2
1, 3
1, 4
2, 3
2, 4
3, 4
```

and 
```go
nums := [5]int{0,1,2,3,4}
for i := 0; i < len(nums) - 2; i++ {
	for j := i+1; j < len(nums) - 1; j++ {
		for k := j+1; k < len(nums); k++ {
			fmt.Printf("%d, %d, %d\n", i, j, k)
		}
	}
}
produces
0, 1, 2
0, 1, 3
0, 1, 4
0, 2, 3
0, 2, 4
0, 3, 4
1, 2, 3
1, 2, 4
1, 3, 4
2, 3, 4
```